packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -192,7 +192,7 @@ class LabeledGlobalKey<T extends State<StatefulWidget>> extends GlobalKey<T> {
   // ignore: prefer_const_constructors_in_immutables , never use const for this class
   LabeledGlobalKey(this._debugLabel) : super.constructor();
 
-  final String? _debugLabel;
+  late final String? _debugLabel;
```

Comment:
I suspect that this field is not going to be tree-shaken as it's a `late final` field without initializer. Such fields are retained because writes to such fields need to guard against repetitive initialization and cannot be removed.

It looks like it doesn't make sense to declare this field `late` as it doesn't have an initializer and it is initialized by constructor.

Reply:
It should be fine to exempt uninitialized debug fields since those won't get initialized in the ctor.

Reply:
At any rate, I believe it's the three fields in this file that are getting initialized iwth an empty map or hashset.

Would it make sense for the compiler to special case these so they don't actually do any work, since we know they don't have side effects?